File: solutions/2554. Maximum Number of Integers to Choose From a Range I/2554.cpp
class Solution {
 public:
  int maxCount(vector<int>& banned, int n, int maxSum) {
    int ans = 0;
    int sum = 0;
    const unordered_set<int> bannedSet{banned.begin(), banned.end()};

    for (int i = 1; i <= n; ++i)
      if (!bannedSet.count(i) && sum + i <= maxSum) {
        ++ans;
        sum += i;
      }

    return ans;
  }
};

File: solutions/2554. Maximum Number of Integers to Choose From a Range I/2554.py
class Solution:
  def maxCount(self, banned: list[int], n: int, maxSum: int) -> int:
    ans = 0
    summ = 0
    bannedSet = set(banned)

    for i in range(1, n + 1):
      if i not in bannedSet and summ + i <= maxSum:
        ans += 1
        summ += i

    return ans
